package/MDAnalysis/analysis/hydrogenbonds/wbridge_analysis.py

```diff
@@ -0,0 +1,1799 @@
+      specified.
+
+        Parameters
+        ----------
+        current : list
+            The current water bridge being analysed is a list of hydrogen bonds
+            from selection 1 to selection 2.
+        output : dict
+            A dictionary which is modified in-place where the key is the type
+            of the water bridge and the value is the number of this type of
+            water  bridge.
+        u : MDAnalysis.universe
+            The current Universe for looking up atoms.
+      '''
+
+      # decompose the first hydrogen bond.
+      sele1_index, sele1_heavy_index, atom2, heavy_atom2, dist, angle =
+      current[0]
+      # decompose the last hydrogen bond.
+      atom1, heavy_atom1, sele2_index, sele2_heavy_index, dist, angle =
+      current[-1]
+      # expand the atom index to the resname, resid, atom names
+      sele1 = u.atoms[sele1_index]
+      sele2 = u.atoms[sele2_index]
+      (s1_resname, s1_resid, s1_name) =
+      (sele1.resname, sele1.resid, sele1.name)
+      (s2_resname, s2_resid, s2_name) =
+      (sele2.resname, sele2.resid, sele2.name)
+
+      # only the residue name is ASP and the atom name is OD1,
+      if s2_resname == 'ASP' and s2_name == 'OD1':
+          # only if the order of water bridge is less than 2
+          if len(current) -1 < 2:
+              # extract all water molecules involved in the water bridge
+              # extract the first water from selection 1
+              s1_index, to_index, (s1_resname, s1_resid, s1_name),
+              (to_resname, to_resid, to_name), dist, angle = current[0]
+              key = (to_resname, to_resid)
+              output[key] = 1
+
+              # extract all the waters between selection 1 and selection 2
+              for hbond in current[1:-1]:
+                  # decompose the hydrogen bond.
+                  from_index, to_index, (from_resname, from_resid, from_name),
+                  (to_resname, to_resid, to_name), dist, angle = hbond
+                  # add first water
+                  key1 = (from_resname, from_resid)
+                  output[key1] = 1
+                  # add second water
+                  key2 = (to_resname, to_resid)
+                  output[key2] = 1
+
+              # extract the last water to selection 2
+              from_index, s2_index, (from_resname, from_resid, from_name),
+              (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+              key = (from_resname, from_resid)
+              output[key] = 1
+
+  w.count_by_time(analysis_func=analysis)
+
+Returns ::
+
+  [(1.0, 1), (2.0, 0),]
+
+Classes
+-------
+
+.. autoclass:: WaterBridgeAnalysis
+   :members:
+
+   .. attribute:: timesteps
+
+      List of the times of each timestep. This can be used together with
+      :attr:`~WaterBridgeAnalysis.timeseries` to find the specific time point
+      of a water bridge existence.
+
+"""
+from __future__ import division
```

Comment:
to ensure proper backwards compatibility we'll need sex.move's range and zip too (worth having a look for other deprecated py2 things, but I _think_ that's it?).

Probably worth adding absolute_import here too.

Reply:
I have added them but given that tests have all passed so I think should be fine.

Reply:
Cheers. Yeah, it's really just for the sake of "correctness".